### PR TITLE
Fixes issue 132 (Public Suffix List (PSL) Updates Too Often)

### LINKED ISSUE #132: Public Suffix List (PSL) Updates Too Often
# 🐛 Summary #

When using the argument '--psl-file=public_suffix.dat', trustymail often updates the PSL when the local version is already up-to-date.

Right now, Trustymail's code logic checks the last modified field of the local PSL using stat.st_mtime, and if the last modified time is more than 24 hours ago, Trustymail downloads a new PSL.

The problem is, the last modified time for a newly downloaded PSL might be 3 days ago. When this is the case, Trustymail downloads a new PSL each time it is run, even though the newest PSL is stored locally.

## To reproduce ##

1. Run `trustymail --psl-file=public_suffix_list.dat cisa.gov`
2. Check to see if the public_suffix_list.dat is more than 24 hours old
3. Repeat step one and verify that an output statement warns, a script is being run to download a new PSL.

## Expected behavior ##

PSL should only be downloaded when the local version is older than that found at publicsuffix.org.

## Any helpful log output or screenshots ##

Paste the results here:

```console

```

Add any screenshots of the problem here.

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #

## 🗣 Description ##
Added code to set the PSL file's last modified time to reflect time of last download.

## 💭 Motivation and context ##
PSL should only be downloaded once per day. The current code attempts to enforce that by only downloading a new PSL, when the local copy is older than 24 hours:
```
psl_age = datetime.now() - datetime.fromtimestamp(
    stat(PublicSuffixListFilename).st_mtime
)
if psl_age > timedelta(hours=24):
    updatePSL(PublicSuffixListFilename)
```
stat.st_mtime checks the last modified date of the local PSL, however, the last modified date will often be 2-4 days ago  even if the PSL was downloaded minutes before.

Resolves #132.

## 🧪 Testing ##
I ran Trustymail against an array of domains.
The first domain was tested and then the code waited for a local PSL to be created before testing the remaining domains. I logged all STDOUT to see if the "PSL Updated" message appeared more than once.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

File: src/trustymail/domain.py [hunk 1]
```diff
@@ -24,12 +24,15 @@ def get_psl():
...
             updatePSL(PublicSuffixListFilename)
+            utime(PublicSuffixListFilename, None) # Set mtime to now
         else:
             psl_age = datetime.now() - datetime.fromtimestamp(
                 stat(PublicSuffixListFilename).st_mtime
             )
             if psl_age > timedelta(hours=24):
                 updatePSL(PublicSuffixListFilename)
+                utime(PublicSuffixListFilename, None) # Set mtime to now
+
```
Comment (jsf9k) [inline src/trustymail/domain.py]: Remove needless blank line:
```suggestion
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I'm requesting one very small change.

File: src/trustymail/domain.py (hunk 1 above)
Comment (jsf9k) [reply src/trustymail/domain.py]: Fixed in commit e8726d8.

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
LGTM 👍

**Comment by jsf9k** [comment]
I also did some testing with these changes and verified that they function as expected.

**Pull request merged**